File: 1.twoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  const map = {};

  for(let i = 0; i < nums.length; i++) {
      const num = nums[i];
      const complement = target - num;

      if(complement in map) {
          return [map[complement], i];
      }
      map[num] = i;
  }

  return [];
};

File: 1.twoSum/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        n = len(nums)

        for i in range(n):
            num = nums[i]
            complement = target - num
            if complement in map:
                return [map[complement], i]
            map[num] = i
        
        return [] # No solution found
